plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftCastPushdown.java

```diff
@@ -0,0 +1,467 @@
+                .addColumn("c_float", "float", asList(1.23, 2.67, null))
+                .addColumn("c_float8", "float8", asList(1.23, 2.67, null))
+                .addColumn("c_decimal_10_2", "decimal(10, 2)", asList(1.23, 2.67, null))
+                .addColumn("c_decimal_19_2", "decimal(19, 2)", asList(1.23, 2.67, null)) // Equal to REDSHIFT_DECIMAL_CUTOFF_PRECISION
+                .addColumn("c_decimal_30_2", "decimal(30, 2)", asList(1.23, 2.67, null))
+                .addColumn("c_numeric_10_2", "numeric(10, 2)", asList(1.23, 2.67, null))
+                .addColumn("c_numeric_19_2", "numeric(19, 2)", asList(1.23, 2.67, null)) // Equal to REDSHIFT_DECIMAL_CUTOFF_PRECISION
+                .addColumn("c_numeric_30_2", "numeric(30, 2)", asList(1.23, 2.67, null))
+                .addColumn("c_char_10", "char(10)", asList("'India'", "'Poland'", null))
+                .addColumn("c_char_50", "char(50)", asList("'India'", "'Poland'", null))
+                .addColumn("c_char_4096", "char(4096)", asList("'India'", "'Poland'", null)) // Equal to REDSHIFT_MAX_CHAR
+                .addColumn("c_nchar_10", "nchar(10)", asList("'India'", "'Poland'", null))
+                .addColumn("c_nchar_50", "nchar(50)", asList("'India'", "'Poland'", null))
+                .addColumn("c_nchar_4096", "nchar(4096)", asList("'India'", "'Poland'", null)) // Equal to REDSHIFT_MAX_CHAR
+                .addColumn("c_bpchar", "bpchar", asList("'India'", "'Poland'", null))
+                .addColumn("c_varchar_10", "varchar(10)", asList("'India'", "'Poland'", null))
+                .addColumn("c_varchar_50", "varchar(50)", asList("'India'", "'Poland'", null))
+                .addColumn("c_varchar_65535", "varchar(65535)", asList("'India'", "'Poland'", null)) // Equal to REDSHIFT_MAX_VARCHAR
+                .addColumn("c_nvarchar_10", "nvarchar(10)", asList("'India'", "'Poland'", null))
+                .addColumn("c_nvarchar_50", "nvarchar(50)", asList("'India'", "'Poland'", null))
+                .addColumn("c_nvarchar_65535", "nvarchar(65535)", asList("'India'", "'Poland'", null)) // Greater than REDSHIFT_MAX_VARCHAR
+                .addColumn("c_text", "text", asList("'India'", "'Poland'", null))
+                .addColumn("c_varbinary", "varbinary", asList("'\\x66696E6465706920726F636B7321'", "'\\x000102f0feee'", null))
+                .addColumn("c_date", "date", asList("DATE '2024-09-08'", "DATE '2019-08-15'", null))
+                .addColumn("c_time", "time", asList("TIME '00:13:42.000000'", "TIME '10:01:17.100000'", null))
+                .addColumn("c_timestamp", "timestamp", asList("TIMESTAMP '2024-09-08 01:02:03.666'", "TIMESTAMP '2019-08-15 09:08:07.333'", null))
+                .addColumn("c_timestamptz", "timestamptz", asList("TIMESTAMP '2024-09-08 01:02:03.666+05:30'", "TIMESTAMP '2019-08-15 09:08:07.333+05:30'", null))
+
+                .addColumn("c_nan", "double precision", asList("'Nan'", "'-Nan'", null))
```

Comment:
NaN is also valid for float as well right ?

Reply:
yes